casper/src/main/scala/coop/rchain/casper/util/comm/DeployRuntime.scala

```diff
@@ -72,32 +73,62 @@ object DeployRuntime {
       phloLimit: Long,
       phloPrice: Long,
       validAfterBlock: Int,
-      maybePrivateKey: Option[PrivateKey],
+      maybePrivateKeyFile: Option[String],
       file: String
-  ): F[Unit] =
-    gracefulExit(
-      Sync[F].delay(Try(Source.fromFile(file).mkString).toEither).flatMap {
-        case Left(ex) =>
-          Sync[F].delay(Left(Seq(s"Error with given file: \n${ex.getMessage}")))
-        case Right(code) =>
+  ): F[Unit] = {
+    val readInputFiles: F[Either[Throwable, (String, Option[PrivateKey])]] =
+      Sync[F]
+        .delay(
           for {
-            timestamp <- Sync[F].delay(System.currentTimeMillis())
-
-            //TODO: allow user to specify their public key
-            d = DeployData()
-              .withTimestamp(timestamp)
-              .withTerm(code)
-              .withPhloLimit(phloLimit)
-              .withPhloPrice(phloPrice)
-              .withValidAfterBlockNumber(validAfterBlock)
-              .withTimestamp(timestamp)
-
-            signedData = maybePrivateKey.fold(d)(SignDeployment.sign(_, d))
-
-            response <- DeployService[F].deploy(signedData)
-          } yield response.map(r => s"Response: $r")
-      }
+            code <- Try(Source.fromFile(file).mkString).toEither
+            maybePrivateKeyBase16 <- maybePrivateKeyFile
+                                      .traverse(f => Try(Source.fromFile(f).mkString).toEither)
+
+            maybePrivateKey <- maybePrivateKeyBase16.traverse(
+                                k =>
+                                  Base16
+                                    .decode(k.toLowerCase)
+                                    .map(PrivateKey(_))
+                                    .toRight(
+                                      new Exception(
+                                        s"Error parsing private key file. Invalid base16 encoding."
+                                      )
+                                    )
+                              )
+          } yield (code, maybePrivateKey)
+        )
+
+    def doDeploy(
+        code: String,
+        maybePrivateKey: Option[PrivateKey]
+    ): F[Either[Seq[String], String]] =
+      for {
+        timestamp <- Sync[F].delay(System.currentTimeMillis())
+
+        //TODO: allow user to specify their public key
+        d = DeployData()
+          .withTimestamp(timestamp)
+          .withTerm(code)
+          .withPhloLimit(phloLimit)
+          .withPhloPrice(phloPrice)
+          .withValidAfterBlockNumber(validAfterBlock)
+          .withTimestamp(timestamp)
+
+        signedData = maybePrivateKey.fold(d)(SignDeployment.sign(_, d))
+
+        response <- DeployService[F].deploy(signedData)
+      } yield response.map(r => s"Response: $r")
+
+    gracefulExit(
+      readInputFiles
+        .flatMap {
+          case Left(ex) =>
+            Sync[F].delay(Left(Seq(s"Error with given file: \n${ex.getMessage}")))
```

Comment:
This is the UI for the end users and we need to provide good error messages. Please create the whole error message in `readInputFiles`. One for the code file, one for the private key file and one for the parsing of the private key from base16.